src/ui/common/src/utils/shared.ts

```diff
@@ -74,6 +75,29 @@ export const getArtifactResultTableRow = (
   };
 };
 
+export function getLatestDagResult(
+  dagResults: DagResultResponse[]
+): DagResultResponse {
```

Comment:
I slightly prefer return an `undefined` here if there's no result available, and let the caller decide how to deal with `undefined` result